### ISSUE TITLE
Expect to revise your models as new (or improved) datasets and methods become available

### ISSUE THREAD

**Issue by biomadeira** [Benjamin-Lee/deep-rules#44]
**Have you checked the [list of proposed rules](https://github.com/Benjamin-Lee/deep-rules/issues?q=is%3Aissue+is%3Aopen+label%3Arule) to see if the rule has already been proposed?**

- [x] Yes

**Feel free to elaborate, rant, and/or ramble.**

Despite all our best efforts in trying to use the the cleanest of the datasets, balanced but heterogenous data, optimal hyperparameter settings, etc., we should expect to revise our models as new data and methods become available. We have to assume that our current set of observations is incomplete and our understanding is limited. Larger and improved datasets, metadata and other annotations will inevitably become available as new experimental techniques are introduced. This is particularly true across prolific fields such as regulatory genomics, transcriptomics, systems biology and biological image analysis (Angermueller et al., 2016, Ching et al., 2018). It will mean that models need to be re-trained and re-tested. Deep learning is still a young technology and in addition to new data, we should expect new developments and improvements to the methods and libraries we currently use.  

**Any citations for the rule?** (peer-reviewed literature preferred but not required)
- Angermueller et al., 2016. [doi.org/10.15252/msb.20156651](https://doi.org/10.15252/msb.20156651)
- Ching et al., 2018. [doi.org/10.1098/rsif.2017.0387](https://doi.org/10.1098/rsif.2017.0387)

**Comment by rasbt** [comment]
Same is maybe true to some extend for architecture improvements. Just a trivial example, but as BatchNorm became kind of established in the field, it could be useful to try adding it to your algorithm and retrain to your data. Or say you used a ResNet-34 architecture and then later came a ResNet-50 with the bottleneck block that might further improve things. Of course, I wouldn't recommend trying every possible new thing out there, but there are several obvious advancements/"tricks" that usually help.

**Comment by fmaguire** [comment]
Mentioned in https://github.com/Benjamin-Lee/deep-rules/blob/master/content/05.dl-complexities.md
